### PR TITLE
Fix refund benchmark for pallet_assets

### PR DESCRIPTION
**Pull request by bkontur**

### CODE REVIEW

**Comment by bkontur** [comment]
bot bench $ pallet dev pallet-assets

File: frame/assets/src/benchmarking.rs [hunk 1]
```diff
@@ -514,8 +514,14 @@ benchmarks_instance_pallet! {
...
-		// `touch` should reserve some balance of the caller...
-		assert!(!T::Currency::reserved_balance(&new_account).is_zero());
+		// check reserved balance of the caller...
+		if T::AssetAccountDeposit::get() > DepositBalanceOf::<T, I>::zero() {
+			// `touch` should reserve some balance of the caller if `AssetAccountDeposit` is required.
+			assert!(!T::Currency::reserved_balance(&new_account).is_zero());
+		} else {
+			// `touch` should not reserve any balance of the caller if no `AssetAccountDeposit` is required.
+			assert!(T::Currency::reserved_balance(&new_account).is_zero());
+		}
```
Comment (joepetrowski) [inline frame/assets/src/benchmarking.rs:524]: I suppose using "not zero" works when there are other deposits being reserved, but in this case maybe it's just better to do:
```suggestion
		assert_eq!(T::Currency::reserved_balance(&new_account), T::AssetAccountDeposit::get());
```

**Review by joepetrowski — APPROVED** [review]

**Comment by bkontur** [comment]
bot rebase

File: frame/assets/src/benchmarking.rs [hunk 2]
```diff
@@ -514,8 +514,7 @@ benchmarks_instance_pallet! {
 			SystemOrigin::Signed(new_account.clone()).into(),
 			asset_id
 		).is_ok());
-		// `touch` should reserve some balance of the caller...
-		assert!(!T::Currency::reserved_balance(&new_account).is_zero());
+		assert_eq!(T::Currency::reserved_balance(&new_account), T::AssetAccountDeposit::get());
```
Comment (bkchr) [inline frame/assets/src/benchmarking.rs:517]: Maybe you can re-add the comment? Basically this pr doesn't "fix" anything, it just ensures that the right amount is reserved or?

Comment (bkontur) [reply frame/assets/src/benchmarking.rs:518]: comment re-added,
yeah, PR fixes just benchmark and right amount comparison, which didnt work for `AssetAccountDeposit = 0`

**Review by ggwpez — APPROVED** [review]

**Pull request merged**